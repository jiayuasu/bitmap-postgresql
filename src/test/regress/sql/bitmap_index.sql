SET enable_seqscan = OFF;
SET enable_indexscan = ON;
SET enable_bitmapscan = ON;

create table bm_test (i int, t text);
insert into bm_test select i % 10, (i % 10)::text  from generate_series(1, 100) i;
create index bm_test_idx on bm_test using bitmap (i);
select count(*) from bm_test where i=1;
select * from bm_test where i > 10;
reindex index bm_test_idx;
select count(*) from bm_test where i=1;
select * from bm_test where i > 10;
drop index bm_test_idx;
drop table bm_test;

