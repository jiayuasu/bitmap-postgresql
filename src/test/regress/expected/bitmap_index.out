SET enable_seqscan = OFF;
SET enable_indexscan = ON;
SET enable_bitmapscan = ON;
create table bm_test (i int, t text);
insert into bm_test select i % 10, (i % 10)::text  from generate_series(1, 100) i;
create index bm_test_idx on bm_test using bitmap (i);
WARNING:  cache reference leak: cache pg_operator (28), tuple 0/17 has count 1
select count(*) from bm_test where i=1;
 count 
-------
    10
(1 row)

select * from bm_test where i > 10;
 i | t 
---+---
(0 rows)

reindex index bm_test_idx;
WARNING:  cache reference leak: cache pg_operator (28), tuple 0/17 has count 1
select count(*) from bm_test where i=1;
 count 
-------
    10
(1 row)

select * from bm_test where i > 10;
 i | t 
---+---
(0 rows)

drop index bm_test_idx;
drop table bm_test;
